coffee-tool/src/main/java/hu/icellmobilsoft/coffee/tool/utils/stream/RequestLoggerInputStream.java

```diff
@@ -0,0 +1,103 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.tool.utils.stream;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.enterprise.inject.Vetoed;
+import javax.enterprise.inject.spi.CDI;
+
+import hu.icellmobilsoft.coffee.tool.utils.string.StringHelper;
+
+/**
+ * Custom {@link InputStream} for logging request with entity
+ *
+ * @author mate.biro
+ * @since 1.15.0
+ */
+@Vetoed
+@SuppressWarnings("InputStreamSlowMultibyteRead")
+public class RequestLoggerInputStream extends InputStream {
+
+    private final InputStream inputStream;
+    private final String requestPrefix;
+    private final StringBuilder entityLog = new StringBuilder();
+    private final StringBuilder logMessage;
+    private int logCollectLimit;
+    private boolean isLogged = false;
+
+    /**
+     * Constructor
+     *
+     * @param requestEntityStream
+     *            original requestEntityStream
+     * @param logCollectLimit
+     *            read limit
+     * @param requestPrefix
+     *            request log prefix
+     * @param logMessage
+     *            log message
+     */
+    public RequestLoggerInputStream(InputStream requestEntityStream, int logCollectLimit, String requestPrefix, StringBuilder logMessage) {
+        this.inputStream = requestEntityStream;
+        this.logCollectLimit = logCollectLimit;
+        this.requestPrefix = requestPrefix;
+        this.logMessage = logMessage;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <br>
+     * Extra functionality: On read the request entity data is appended from the original {@link InputStream} to an internal {@link StringBuilder}
+     * until a given limit is reached (or until the end of stream if log limit exceeds the length of the stream). Then publishes an event to log the
+     * given request message with the appended request entity.
+     *
+     */
+    @Override
+    public int read() throws IOException {
+        int streamData = inputStream.read();
+
+        if (streamData != -1 && logCollectLimit != 0) {
+            // logoláshoz gyűjtjük a stream tartalmát amíg van, és még nem értük el a limitet
+            entityLog.append((char) streamData);
+            logCollectLimit--;
+        } else if (!isLogged) {
+            // ha a stream végére értünk vagy elértük a limitet és még nem logoltunk,
+            // akkor küldünk egy eventet hogy megtörténjen a logolás
+            String maskedEntityLog = getMaskedEntity(entityLog.toString());
+            logMessage.append(maskedEntityLog);
+
+            LoggingEvent event = new LoggingEvent(logMessage.toString());
+            LoggingPublisher loggingPublisher = CDI.current().select(LoggingPublisher.class).get();
```

Comment:
Ezen a ponton mivel ez dependent minden alkalommal letrehoz egy uj instancet, ApplicationScope hatékonyabb lenne. Egyébként a fire nélkül itt mar logolhatná is ki egyből, hol ad plust az a fire? Mem leaket nem talaltam igy sem hogy Dependent

Reply:
Átraktam ApplicationScope-ra.

Azért van event dobva a logolásra, hogy azt ne konkrétan ez a stream osztály végezze. Azon kívül, hogy így jobban szeparálva van a logika, más előnyt nem látok én sem.

https://github.com/i-Cell-Mobilsoft-Open-Source/coffee/pull/376#discussion_r1081077545

Reply:
ok, egy class siman injectel is ugyan ez lenne, fire nelkul, nekem igy is jo, threadet jol kezeli, részemről okes